spec/lib/synapse/service_watcher_zookeeper_spec.rb

```diff
@@ -155,13 +117,103 @@
         ["test_child_1"]
       )
       expect(mock_zk).to receive(:get).with('some/path', {:watch=>true}).and_return("")
-      expect(mock_zk).to receive(:get).with('some/path/test_child_1', {}).and_raise(ZK::Exceptions::NoNode)
+      expect(mock_zk).to receive(:get).with('some/path/test_child_1').and_raise(ZK::Exceptions::NoNode)
 
       subject.instance_variable_set('@zk', mock_zk)
       expect(subject).to receive(:set_backends).with([],{})
       subject.send(:watcher_callback).call
     end
 
+    context 'retry watcher_callback' do
```

Comment:
In Rspec, you use `describe` to label *what* you are testing, `context` to label *when* that test applies (i.e. the conditions that it checks) and `it` to label the *expectation*. 

In this case, you want to use `describe watcher_callback`, `context 'when it fails'`, `it 'retries until success'` or something like that. While that doesn't change the behavior of the tests, it does make the tests easier to read :)

Reply:
right now most of existing tests use `describe` to label which class to test, `context` to label which `method`, and `it` to label both condition and expectation.

agree this is not ideal, but at least I can make it consistent with existing tests.

Reply:
Okay, that's fair